### PR TITLE
adds jobs to build diamond

### PR DESCRIPTION
**Pull request by GregMeno**
@ktdreyer 
@alfredodeza 

Signed-off-by: Gregory Meno gmeno@redhat.com

### CODE REVIEW

File: diamond-build/config/definitions/diamond [hunk 1]
```diff
@@ -0,0 +1,48 @@
+- job:
+    name: diamond-build
+    node: small && trusty
```
Comment (alfredodeza) [inline diamond-build/config/definitions/diamond]: would you mind doing the following to be a bit more flexible:

    small && (trusty || centos)

File: diamond-docs/build/build [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Define and ensure the PIP cache
+PIP_SDIST_INDEX="$HOME/.cache/pip"
+mkdir -p $PIP_SDIST_INDEX
+
+# Install the package by trying with the cache first, otherwise doing a download only, and then
+# trying to install from the cache again.
+if ! venv/bin/pip install --find-links="file://$PIP_SDIST_INDEX" --no-index tox; then
+    venv/bin/pip install --download-directory="$PIP_SDIST_INDEX" tox
+    venv/bin/pip install --find-links="file://$PIP_SDIST_INDEX" --no-index tox
+fi
```
Comment (alfredodeza) [inline diamond-docs/build/build]: most of the contents on this script are already available for you if you are pulling it in the shell builder with:

    - ../../../scripts/build_utils.sh

If you do that, then this script can be like the ceph-installer: https://github.com/ceph/ceph-build/blob/master/ceph-installer-docs/build/build

File: diamond-tag/config/definitions/ceph-tag.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+- scm:
```
Comment (alfredodeza) [inline diamond-tag/config/definitions/ceph-tag.yml]: this file is named ceph-tag.yml ?

File: diamond-tag/config/definitions/ceph-tag.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    name: ceph-build
```
Comment (alfredodeza) [inline diamond-tag/config/definitions/ceph-tag.yml]: this should not be ceph-build

File: diamond/config/definitions/diamond [hunk 5]
```diff
@@ -0,0 +1,62 @@
+- job:
```
Comment (alfredodeza) [inline diamond/config/definitions/diamond]: this file should probably be: `diamond.yml`

File: diamond-build/build/build [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+if listcontains $DIST "$rpm_dists"
+then
+        # Tag tree and update version number in change log and
+        # in setup.py before building.
+
+        # this exists in scripts/build_utils.sh
+        get_rpm_dist
+
+        REPO=rpm-repo
```
Comment (alfredodeza) [inline diamond-build/build/build]: I don't see this used anywhere else (same thing in the DEB section)

**Comment by GregMeno** [comment]
@alfredodeza this is just a straight search and replace of ceph-deploy. thank you for the feedback.

**Pull request closed without merging**